src/main/java/rx/internal/util/BlockingUtils.java

```diff
@@ -0,0 +1,38 @@
+package rx.internal.util;
+
+import rx.Subscription;
+
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Utility functions relating to blocking types.
+ *
+ * Not intended to be part of the public API.
+ */
+public class BlockingUtils {
+
+	/**
+	 * Block and wait for a {@link Subscription} to complete.
+	 *
+	 * @param latch a CountDownLatch
+	 * @param subscription the Subscription to wait on.
+	 */
+	public static void awaitForComplete(CountDownLatch latch, Subscription subscription) {
```

Comment:
An identical function (`awaitForComplete`) is also used in `BlockingObservable`. Worth modifying that class to call this function to avoid duplication?